microsoft-edge/devtools-guide-chromium/memory-problems/dom-leaks-memory-tool-detached-elements.md

```diff
@@ -0,0 +1,144 @@
+
+
+<!-- ====================================================================== -->
+## About DOM leaks due to detached elements
+
+To increase the performance of your webpage, find detached elements that the browser can't garbage-collect, and then locate the JavaScript object that's still referencing the detached element.  Then change your JavaScript to release the element, to reduce the number of detached elements on your webpage, increasing webpage performance and responsiveness.
+<!-- copied from dom-leaks.md -->
+
+
+<!-- copied from dom-leaks.md -->
+<!-- ------------------------------ -->
+#### Fixing the JavaScript code to prevent detached elements that can't be garbage-collected
+
+In the demo webpage, the `collectOldMessages` method of the `Room` class (on line 26) contains comments about a potential leak, and recommends making a single cache for all `Room` instances, instead of a cache for each `Room` instance, and then monitoring how many elements get into that cache over time.
+
+There is a potential leak in the existing code, because the cleanup in this method occurs at a different rate than the addition of new messages.  We can easily find ourselves in a situation where we have messages in the cache that don't need (or can't) be reused right away.  This is not likely to become too much, but the cache is per-room, which multiplies the problem.
+
+
+<!-- copied from dom-leaks.md, & trimmed to omit DE tool's UI controls -->
+<!-- ====================================================================== -->
+## About detached elements and memory leaks
+
+Detached elements aren't always an indication of a memory leak, and memory leaks aren't always caused by detached elements.  Memory leaks depend on the context of your application.
+
+A memory leak can occur in your application when an element is no longer attached to the Document Object Model (DOM) tree, but is still referenced by some JavaScript running on the webpage. These elements are called *detached elements*.  For the browser to garbage-collect (GC) the detached element, the element must not be referenced from the DOM tree or from JavaScript code.
+
+Memory issues affect webpage performance, including memory leaks, memory bloat, and frequent garbage collections.  Symptoms for your users include:
+
+*  The performance of a webpage gets progressively worse over time.
+*  The performance of a webpage is consistently bad.
+*  The performance of a webpage is delayed or appears to pause frequently.
+
+
+<!-- copied (& trimmed) from dom-leaks.md -->
+<!-- ====================================================================== -->
+## Re-attaching elements
+
+For this demo webpage, it makes sense to retain the list of chat messages, so that if a user switches back to **Room 1**, the message log is preserved.  Similarly, a feed in social media might detach elements as users scroll past them, and reattach them to the DOM when users scroll back up.
```

Comment:
added:
When the **Detached elements** tool reports detached elements, it's not necessarily a memory leak.  It's up to you to decide what's a memory leak and what's not.  Maybe your app will re-attach those elements later (instead of having to re-create them, which could be slower).

Detaching DOM nodes is a useful approach, as long as you eventually reuse those detached elements (or delete them).  The value of the **Detached elements** tool is, when you suspect a memory leak, you can check whether there are an increasing number of unexpected detached DOM elements being reported by the tool.

Reply:
section About detached elements and memory leaks --
no change needed in new article, b/c didn't add the 5 extra paragraphs below the list, when copying from DE tool article